map_test.go

```diff
@@ -182,3 +182,42 @@ func TestJavaMap(t *testing.T) {
 	RegisterPOJO(customMap)
 	testJavaDecode(t, "customArgTypedFixed_CustomMap", customMap)
 }
+
+type Obj struct {
+	Map8  map[int8]int8
+	Map16 map[int16]int16
+	Map32 map[int32]int32
+}
+
+func (Obj) JavaClassName() string {
+	return ""
+}
+
+func TestMapInObject(t *testing.T) {
+	var (
+		obj Obj
+		e   *Encoder
+		d   *Decoder
+		err error
+		res interface{}
+	)
+
+	obj = Obj{
+		Map8:  map[int8]int8{1: 2, 3: 4},
+		Map16: map[int16]int16{1: 2, 3: 4},
+		Map32: map[int32]int32{1: 2, 3: 4},
+	}
+
+	e = NewEncoder()
+	e.Encode(obj)
+	if len(e.Buffer()) == 0 {
+		t.Fail()
+	}
+
+	d = NewDecoder(e.Buffer())
+	res, err = d.Decode()
+	if err != nil {
+		t.Errorf("Decode() = %+v", err)
+	}
+	t.Logf("decode(%v) = %v, %v\n", obj, res, err)
```

Comment:
should check whether the decode value is equal to the original

Reply:
done